src/ape/api/projects.py

```diff
@@ -0,0 +1,229 @@
+import json
+import shutil
+from pathlib import Path
+from typing import TYPE_CHECKING, ClassVar, Collection, Dict, Optional
+
+from ethpm_types import Checksum, ContractType, PackageManifest, Source
+from ethpm_types.manifest import PackageName
+from ethpm_types.utils import compute_checksum
+
+from ape.exceptions import ProjectError
+from ape.logging import logger
+from ape.types import DependencyType
+from ape.utils import (
+    abstractmethod,
+    get_all_files_in_directory,
+    get_relative_path,
+    injected_before_use,
+)
+
+if TYPE_CHECKING:
+    from ape.managers.compilers import CompilerManager
+
+
+class ProjectAPI:
```

Comment:
At this time, I am not sure what to do as far as making this some sort of BaseModel. Thoughts?

Reply:
Seems like `path` would be a good thing to do that for. Right now `path` is a class variable, and this will probably not work too well when you load multiple projects.